landing_page/lib/tasks/archive_posts.rake

```diff
@@ -0,0 +1,12 @@
+desc "Archive posts periodically based on if they are one week old or more"
+
+task archive: :environment do
+    archiveable_posts = Micropost.order(created_at: :desc).kept.where("created_at <= ?", Time.now-7.days)
+
+    archiveable_posts.each do |archiveable_post|
+        archiveable_post.archived = true
+        archiveable_post.save
+    end
+
+    puts "Number of posts to be archived is: #{archiveable_posts.count}"
```

Comment:
Given the way this message reads, it should be placed **before** the loop.

Reply:
Resolved in commit 9751f363583802be9460bcdbac2d3b9ee2237c12